File: 파이썬/구명보트.py
def solution(people, limit):
    N = len(people)
    together, right = 0, N - 1
    people.sort()  # 정렬
    for left in range(N):
        while left < right:
            if people[left] + people[right] <= limit:
                together += 1
                right -= 1
                break
            right -= 1
    return N - together

File: 자바스크립트/구명보트.js
function solution(people, limit) {
    people.sort((a, b) => a - b);
    const number = people.length;
    let together = 0;
    let right = number - 1;
    for (let left = 0; left < number; left++) {
        while (left < right) {
            if (people[left] + people[right] <= limit) {
                together++;
                right--;
                break;
            }
            right--
        }
    }
    return number - together;
}